axum-extra/src/routing/typed.rs

```diff
@@ -295,3 +382,63 @@ impl_second_element_is!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13);
 impl_second_element_is!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14);
 impl_second_element_is!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15);
 impl_second_element_is!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    struct UsersShow {
+        id: i32,
+    }
+
+    // can't derive `TypedPath` because that references `::axum_extra` :(
+    impl fmt::Display for UsersShow {
```

Comment:
Can this test module not be moved into a separate file where both could be imported without creating a cycle?

Reply:
No. Its not because there is a cycle but because the generated code contains `::axum_extra::routing::TypedPath`, i.e. it access axum_extra from the root. I don't think its a big deal for this test.

Reply:
🤷👍

Reply:
We can solve this by adding `extern crate self as axum_extra;` to the crate root. Then the test can use the proc-macro and will be closer to a real-world scenario.

Reply:
TIL!